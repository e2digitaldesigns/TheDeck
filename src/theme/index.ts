export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
