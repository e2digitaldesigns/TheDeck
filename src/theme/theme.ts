export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
