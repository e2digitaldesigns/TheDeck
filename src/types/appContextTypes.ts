export interface IntAppDataIconSelector {
  isVisible: boolean;
  icon: string;
}

export interface IntAppDataActive {
  profileId: string;
  pageId: string;
  buttonPadId: string;
  actionId: string;
}

export interface IntDropZones {
  actionList: boolean;
  buttonPads: boolean;
  sideBarProfiles: boolean;
  sideBarStyles: boolean;
  styleHeader: boolean;
}

export interface IntAppContextInterface {
  iconSelector: IntAppDataIconSelector;
  active: IntAppDataActive;
  dropZones: IntDropZones;
}

export interface IntAppContextStateInterface {
  appState: IntAppContextInterface;
  setAppState: React.Dispatch<React.SetStateAction<any>>;
}

export type IntAppData = IntAppContextStateInterface;

export enum AppStateActive {
  ProfileId = "profileId",
  PageId = "pageId",
  ButtonPadId = "buttonPadId",
  ActionId = "actionId"
}
