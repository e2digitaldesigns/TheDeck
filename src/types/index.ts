/* istanbul ignore file */
export * from "./actionType";
export * from "./aplicationTypes";
export * from "./appContextTypes";
export * from "./buttonPadGridTypes";
export * from "./dragAndDropTypes";
export * from "./globalContextType";
export * from "./globalType";
export * from "./ipcRenderTypes";
export * from "./obsTypes";
