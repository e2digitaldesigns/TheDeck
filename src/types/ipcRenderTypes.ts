export enum IpcRendererTypes {
  database = "database",
  databaseReturn = "database:return",
  header = "header",
  importSettings = "importSettings",
  importSettingsReturn = "importSettings:return",
  mdPlaySound = "MD:playSound"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData"
}
