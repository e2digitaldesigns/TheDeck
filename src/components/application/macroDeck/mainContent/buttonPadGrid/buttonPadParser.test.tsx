import "@testing-library/jest-dom";
import { cleanup, fireEvent, render } from "@testing-library/react";
import _cloneDeep from "lodash/cloneDeep";
import ButtonPadParser from "./buttonPadParser";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../theme/theme";

import initAppState from "../../../../../__mock_data__/appState.json";
import initDropZoneState from "../../../../../__mock_data__/dropZoneState.json";
import initGlobalState from "../../../../../__mock_data__/globalState.json";

import { IntAppData, IntDropZones, IntGlobalData } from "../../../../../types";

const mockAppState: IntAppData = {
  appState: _cloneDeep(initAppState),
  setAppState: jest.fn(appState => appState)
};

const mockDropZoneState: any = {
  dropZoneState: _cloneDeep(initDropZoneState),
  setDropZoneState: jest.fn(dropZoneState => dropZoneState)
};

const mockGlobalState: IntGlobalData = {
  state: _cloneDeep(initGlobalState),
  setState: jest.fn(state => state)
};

const mockHooks = {
  useAppData: mockAppState,
  useDropZone: mockDropZoneState,
  useGlobalData: mockGlobalState,
  activateButtonPad: jest.fn(),
  createButtonPad: jest.fn(),
  copyState: {
    buttonPad: {
      bgColor: "buttonPad.bgColor",
      icon: "buttonPad.icon",
      iconColor: "buttonPad.iconColor",
      text: "buttonPad.text",
      textColor: "buttonPad.textColor"
    },
    actions: []
  },
  deleteButtonPad: jest.fn(),
  handleButtonCopy: jest.fn(),
  pasteButtonPad: jest.fn(),
  readButtonPad: () => mockGlobalState.state.buttonPads[0]
};

jest.mock("../../../../../hooks", () => ({
  useAppData: () => mockHooks.useAppData,
  useButton: () => ({
    activateButtonPad: mockHooks.activateButtonPad,
    createButtonPad: mockHooks.createButtonPad,
    deleteButtonPad: mockHooks.deleteButtonPad,
    pasteButtonPad: mockHooks.pasteButtonPad,
    readButtonPad: mockHooks.readButtonPad
  }),
  useDropZone: () => mockHooks.useDropZone,
  useGlobalData: () => mockHooks.useGlobalData,
  useProfile: () => ({
    readProfile: () => mockGlobalState.state.profiles[0]
  })
}));

const setup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <ButtonPadParser
        copyState={mockHooks.copyState}
        handleButtonCopy={mockHooks.handleButtonCopy}
        padNumber={1}
      />
    </ThemeProvider>
  );
};

beforeEach(() => {
  mockAppState.appState = _cloneDeep(initAppState);
  mockDropZoneState.dropZoneState = _cloneDeep(initDropZoneState);
  mockGlobalState.state = _cloneDeep(initGlobalState);
});

afterEach(() => {
  jest.resetAllMocks();
  jest.clearAllMocks();
  cleanup;
});

describe("<ButtonPadParser/>", () => {
  it("Should render correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.getByTestId("button_pad_parser__button-inner")).toBeTruthy();
  });

  it("Should call deleteButtonPad", () => {
    const wrapper = setup();
    const component = wrapper.getByTestId("button_pad_parser__option-delete");
    expect(component).toBeTruthy();

    fireEvent.click(component);
    expect(mockHooks.deleteButtonPad).toHaveBeenCalledTimes(1);
    expect(mockHooks.deleteButtonPad).toHaveBeenCalledWith(
      mockGlobalState.state.buttonPads[0]._id
    );
  });

  it("Should call activateButtonPad", () => {
    const wrapper = setup();
    const component = wrapper.getByTestId("button_pad_parser__option-edit");
    expect(component).toBeTruthy();

    fireEvent.click(component);
    expect(mockHooks.activateButtonPad).toHaveBeenCalledTimes(1);
    expect(mockHooks.activateButtonPad).toHaveBeenCalledWith(
      mockGlobalState.state.buttonPads[0]._id
    );
  });

  it("Should call activateButtonPad", () => {
    mockGlobalState.state.buttonPads = [];
    const wrapper = setup();
    const component = wrapper.getByTestId("button_pad_parser__create");
    expect(component).toBeTruthy();

    fireEvent.click(component);
    expect(mockHooks.createButtonPad).toHaveBeenCalledTimes(1);
    expect(mockHooks.createButtonPad).toHaveBeenCalledWith(1);
  });
});

describe("<ButtonPadParser/> Drag & Drop", () => {
  it("Should start drag", () => {
    const wrapper = setup();
    const newDropZoneState = _cloneDeep(mockDropZoneState.dropZoneState);
    const buttonPad = wrapper.getByTestId("button_pad_parser__button-inner");

    expect(buttonPad).toBeTruthy();
    expect(buttonPad).toHaveAttribute("draggable");

    fireEvent.dragStart(buttonPad);
    newDropZoneState.dropZones.buttonPads = true;
    newDropZoneState.dropZones.styleHeader = true;

    expect(mockDropZoneState.setDropZoneState).toHaveBeenCalledTimes(1);
    expect(mockDropZoneState.setDropZoneState).toHaveBeenCalledWith(
      newDropZoneState
    );

    fireEvent.dragEnd(buttonPad);
    newDropZoneState.dropZones.buttonPads = false;
    newDropZoneState.dropZones.styleHeader = false;
    expect(mockDropZoneState.setDropZoneState).toHaveBeenCalledTimes(2);
    expect(mockDropZoneState.setDropZoneState).toHaveBeenCalledWith(
      newDropZoneState
    );
  });
});

describe("<ButtonPadParser/> Copy & Paste", () => {
  it("Should copy button", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;

    const wrapper = setup();
    const copyButton = wrapper.getByTestId("button_pad_parser__option-copy");
    expect(copyButton).toBeTruthy();

    fireEvent.click(copyButton);
    expect(mockHooks.handleButtonCopy).toHaveBeenCalledTimes(1);
    expect(mockHooks.handleButtonCopy).toHaveBeenCalledWith(
      mockGlobalState.state.buttonPads[0]
    );
  });

  it("Should paste button", () => {
    mockAppState.appState.active.buttonPadId =
      mockGlobalState.state.buttonPads[0]._id;

    const wrapper = setup();
    const pasteButton = wrapper.getByTestId("button_pad_parser__option-paste");
    expect(pasteButton).toBeTruthy();

    fireEvent.click(pasteButton);
    expect(mockHooks.pasteButtonPad).toHaveBeenCalledTimes(1);
    expect(mockHooks.pasteButtonPad).toHaveBeenCalledWith(
      mockGlobalState.state.buttonPads[0],
      mockHooks.copyState
    );
  });
});
