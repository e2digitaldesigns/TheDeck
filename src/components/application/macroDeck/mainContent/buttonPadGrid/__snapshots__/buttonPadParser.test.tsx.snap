// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPadParser/> Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsDKAQ dojDCw"
        data-testid="button_pad_parser__button-inner"
        draggable="true"
      >
        <div
          class="sc-dkPtRN sc-hKwDye cQfquZ bHjDpw"
          data-testid="button_pad_parser__option-delete"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </div>
        <div
          class="sc-dkPtRN sc-eCImPb cQfquZ ZYwUD"
          data-testid="button_pad_parser__option-edit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </div>
        <div
          class="sc-dkPtRN sc-jRQBWg cQfquZ laMsOI"
          data-testid="button_pad_parser__option-copy"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
            />
          </svg>
        </div>
        <div
          class="sc-dkPtRN sc-gKclnd cQfquZ idNYlO"
          data-testid="button_pad_parser__option-paste"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
            />
          </svg>
        </div>
        <div
          class="sc-furwcr cJNJun"
        >
          <svg
            data-testid="icon-Image"
            fill="none"
            height="32"
            stroke="#cccccc"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="18"
              rx="2"
              ry="2"
              width="18"
              x="3"
              y="3"
            />
            <circle
              cx="8.5"
              cy="8.5"
              r="1.5"
            />
            <polyline
              points="21 15 16 10 5 21"
            />
          </svg>
        </div>
        <div
          class="sc-iCfMLu edopZs"
          color="#cccccc"
        >
          Working
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDKAQ dojDCw"
      data-testid="button_pad_parser__button-inner"
      draggable="true"
    >
      <div
        class="sc-dkPtRN sc-hKwDye cQfquZ bHjDpw"
        data-testid="button_pad_parser__option-delete"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </div>
      <div
        class="sc-dkPtRN sc-eCImPb cQfquZ ZYwUD"
        data-testid="button_pad_parser__option-edit"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </div>
      <div
        class="sc-dkPtRN sc-jRQBWg cQfquZ laMsOI"
        data-testid="button_pad_parser__option-copy"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
          />
        </svg>
      </div>
      <div
        class="sc-dkPtRN sc-gKclnd cQfquZ idNYlO"
        data-testid="button_pad_parser__option-paste"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
          />
        </svg>
      </div>
      <div
        class="sc-furwcr cJNJun"
      >
        <svg
          data-testid="icon-Image"
          fill="none"
          height="32"
          stroke="#cccccc"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="18"
            rx="2"
            ry="2"
            width="18"
            x="3"
            y="3"
          />
          <circle
            cx="8.5"
            cy="8.5"
            r="1.5"
          />
          <polyline
            points="21 15 16 10 5 21"
          />
        </svg>
      </div>
      <div
        class="sc-iCfMLu edopZs"
        color="#cccccc"
      >
        Working
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
